File: src/main/java/br/com/brforgers/ducts/screens/DuctScreen.java
package br.com.brforgers.ducts.screens;

import javax.annotation.Nonnull;

import com.mojang.blaze3d.systems.RenderSystem;
import com.mojang.blaze3d.vertex.PoseStack;

import br.com.brforgers.ducts.Ducts;
import br.com.brforgers.ducts.inventories.DuctInventory;
import net.minecraft.client.gui.screens.inventory.AbstractContainerScreen;
import net.minecraft.client.renderer.GameRenderer;
import net.minecraft.network.chat.Component;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.world.entity.player.Inventory;
import net.minecraftforge.api.distmarker.Dist;
import net.minecraftforge.api.distmarker.OnlyIn;

@OnlyIn(Dist.CLIENT)
public class DuctScreen extends AbstractContainerScreen<DuctInventory>
{
	private static final ResourceLocation HOPPER_GUI_TEXTURE = new ResourceLocation(Ducts.MOD_ID, "textures/gui/container/duct.png");
	
	public DuctScreen(DuctInventory pMenu, Inventory pPlayerInventory, Component pTitle) 
	{
		super(pMenu, pPlayerInventory, pTitle);
		
		passEvents = false;
		imageHeight = 133;
		inventoryLabelY = imageHeight - 94;
	}
	
	@Override
	public void render(@Nonnull PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) 
	{
        this.renderBackground(matrixStack);
        super.render(matrixStack, mouseX, mouseY, partialTicks);
        this.renderTooltip(matrixStack, mouseX, mouseY);
    }

	@Override
	protected void renderBg(PoseStack matrixStack, float pPartialTick, int pMouseX, int pMouseY) 
	{
		if (minecraft != null) 
		{
            RenderSystem.setShader(GameRenderer::getPositionTexShader);
            RenderSystem.setShaderColor(1.0f, 1.0f, 1.0f, 1.0f);
            RenderSystem.setShaderTexture(0, HOPPER_GUI_TEXTURE);
            int i = (width - imageWidth) / 2;
            int j = (height - imageHeight) / 2;
            this.blit(matrixStack, i, j, 0, 0, imageWidth, imageHeight);
        }
	}
}

File: src/main/kotlin/br/com/brforgers/mods/ducts/screens/DuctScreen.kt
package br.com.brforgers.mods.ducts.screens

import br.com.brforgers.mods.ducts.Ducts
import br.com.brforgers.mods.ducts.blockentities.DuctBlockEntity
import br.com.brforgers.mods.ducts.inventories.DuctInventory
import com.mojang.blaze3d.matrix.MatrixStack
import com.mojang.blaze3d.systems.RenderSystem
import net.minecraft.client.gui.screen.inventory.ContainerScreen
import net.minecraft.entity.player.PlayerInventory
import net.minecraft.util.ResourceLocation
import net.minecraft.util.text.ITextComponent
import net.minecraftforge.api.distmarker.Dist
import net.minecraftforge.api.distmarker.OnlyIn
import javax.annotation.Nonnull


@OnlyIn(Dist.CLIENT)
class DuctScreen(screenContainer: DuctInventory, inv: PlayerInventory, title: ITextComponent) :
    ContainerScreen<DuctInventory>(screenContainer, inv, title) {
    private val duct: DuctBlockEntity

    override fun render(@Nonnull matrixStack: MatrixStack, mouseX: Int, mouseY: Int, partialTicks: Float) {
        this.renderBackground(matrixStack)
        super.render(matrixStack, mouseX, mouseY, partialTicks)
        this.renderTooltip(matrixStack, mouseX, mouseY)
    }

    override fun renderBg(@Nonnull matrixStack: MatrixStack, partialTicks: Float, x: Int, y: Int) {
        if (minecraft != null) {
            RenderSystem.color4f(1f, 1f, 1f, 1f)
            minecraft!!.textureManager.bind(HOPPER_GUI_TEXTURE)
            val i = (width - imageWidth) / 2
            val j = (height - imageHeight) / 2
            this.blit(matrixStack, i, j, 0, 0, imageWidth, imageHeight)
        }
    }

    companion object {
        private val HOPPER_GUI_TEXTURE =
            ResourceLocation(Ducts.ID, "textures/gui/container/duct.png")
    }

    init {
        duct = screenContainer.duct
        passEvents = false
        imageHeight = 133
        inventoryLabelY = imageHeight - 94
    }
}